src/gui/qgsscrollarea.cpp

```diff
@@ -0,0 +1,110 @@
+/***************************************************************************
+    qgspanelwidget.cpp
```

Comment:
Please don't let the general crappiness of the coding in this PR distract from the bigger question. I deleted the PR guideline checklist for a reason ;)